extensions/amp-auto-lightbox/0.1/carousel-criteria.js

```diff
@@ -0,0 +1,294 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {Layout} from '../../../src/layout';
```

Comment:
From what I see, when amp-lightbox-gallery processes a carousel, it checks its slides against what's allowed (img and figure), so I am not sure how this actually works.

Aside from the sizing criterial, every other check should be shared with amp-lightbox-gallery code.

Reply:
My main point is `lightbox` attribute on `amp-carousel` itself is an auto lightbox feature. Currently its heuristics are basic and if anything fails the heuristics it errors. 

If we relax those conditions here, we should also relax them there. Only difference is whether "error" is produced. If developer explicitly puts `lightbox` on `amp-carousel`, then error is warranted.

so ideally we have something like this:
```
file: lightbox-gallery.js
isCarouselLightboxable(carousel){...}

if(carousel && isCarouselLightboxable(carousel)) {
  // make it lightboxable
} else {
error
}

file: auto-lightbox.js
if(caroucel && LightboxGallery.isCarouselLightboxable(carousel)) {
 carousel.addAttribute('lightbox');
 // gauranteed to work
}
```

/cc @cathyxz for thoughts